android/app/src/main/java/org/fossasia/openevent/utils/ISO8601Date.java

```diff
@@ -51,12 +51,12 @@ public static String dateFromCalendar(Calendar currentDate) {
     }
 
     public static String getTimeZoneDateFromString(String dateString) throws ParseException {
-        SimpleDateFormat formatter = new SimpleDateFormat("yyyy-MM-DD", Locale.getDefault());
+        SimpleDateFormat formatter = new SimpleDateFormat("yyyy-MM-dd", Locale.getDefault());
         formatter.setTimeZone(getEventTimezone());
         Date date = formatter.parse(dateString);
-        SimpleDateFormat newFormat = new SimpleDateFormat("d MMM", Locale.getDefault());
-
-        return newFormat.format(date);
+        SimpleDateFormat newFormat = new SimpleDateFormat("d MMM, z", Locale.getDefault());
```

Comment:
Yes i have checked it on my device. Without 'z' it is formatting date with "GMT+5:30" for both time zones that's why i am using 'z'

Just check it using Log in getTimeZoneDateFromString() function
`dateString +"  "+ date.toString()`